geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/command/Ping.java

```diff
@@ -55,12 +55,11 @@ public void cmdExecute(final Message clientMessage, final ServerConnection serve
             (InternalDistributedMember) clientMessage.getPart(0).getObject();
         InternalDistributedMember myID = serverConnection.getCache().getMyId();
         if (!myID.equals(targetServer)) {
-          if (myID.compareTo(targetServer.getMemberIdentifier(), true, false) == 0) {
+          if (myID.compareTo(targetServer, true, false) == 0) {
             logger.warn("Target server {} has different viewId {}", targetServer, myID);
             writeErrorResponse(clientMessage, MessageType.EXCEPTION, serverConnection);
           } else {
             pingCorrectServer(clientMessage, targetServer, serverConnection);
-            writeReply(clientMessage, serverConnection);
```

Comment:
@jujoramos Implementing tests I realized this writeReply here is duplicated, `pingCorrectServer` is already calling it when the ping is forwarded. So this should be the cause for the error you saw about an unexpected REPLY message.